mobile/lib/interfaces/folder_api.interface.dart

```diff
@@ -0,0 +1,7 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/entities/asset.entity.dart';
+
+abstract interface class IFolderApiRepository {
+  Future<AsyncValue<List<String>>> getAllUniquePaths();
```

Comment:
We should just return the `Future<List<String>>` here. So that we don't add additional type dependency from `riverpod` library to the repository